### PR TITLE
Update user input

### PR DESCRIPTION
**Pull request by anrattanasouk**
- Successfully updated database through user input from CreateActivityView (title, category, options, (might leave off friends tagging in beta release)
- Having trouble with 
   - getting/passing in activity id and option id (using hardcoded activity id and options id for now)
   - looping through the array of `var options: [Option.ID: PollOption] = [:]` in Activity
- Important TODO is to have the HomeView loop through the `var activities: [Activity.ID] = []` in User

### CODE REVIEW

File: Shared/View/VoteView.swift [hunk 1]
```diff
@@ -8,28 +8,50 @@
...
-        id: "123",
-        title: "What movie should we watch tonight?",
-        category: Category.movie,
-        // FIXME: this is not a valid user ID
-        // TODO: request renaming by backend if this is confusing
-        author: "mightyDonut"
-    )
+    @EnvironmentObject var user: UserViewModel
+    @EnvironmentObject var activity: ActivityViewModel
+    @EnvironmentObject var option: OptionViewModel
```
Comment (ApolloZhu) [inline Shared/View/VoteView.swift:13]: These two are not really suitable to pass down as environment object. They are more so designed to observe a specific activity. So instead you'll have something like 

```swift
@ObservedObject var activityViewModel = ActivityViewModel(id: "AhdjSLOlTJqb68aXBAWn") 
```

then you can get an always up to date activity with

```swift
activityViewModel.activity
```

Comment (ApolloZhu) [inline Shared/View/VoteView.swift:13]: These two are not really suitable to pass down as environment object. They are more so designed to observe a specific activity. So instead you'll have something like 

```swift
@ObservedObject var activityViewModel = ActivityViewModel(id: "AhdjSLOlTJqb68aXBAWn") 
```

then you can get an always up to date activity with

```swift
activityViewModel.activity
```

File: Shared/View/VoteView.swift [hunk 2]
```diff
@@ -8,28 +8,50 @@
...
+                        act = try await ActivityViewModel.activity(id: "AhdjSLOlTJqb68aXBAWn")
+                    } catch {
+                        print(error)
+                    }
+                }
+            } label: {
+                NavigationLink(destination: VoteView()) {
+                    Text("Get Activity")
+                }
+            }
```
Comment (ApolloZhu) [inline Shared/View/VoteView.swift:39]: ... then you probably won't need this

File: Shared/View/VoteView.swift [hunk 3]
```diff
@@ -8,28 +8,50 @@
...
+            // Text("activityid \(activity.activity?.id ?? "")")
+            
+            Text("Activity \(act.title)").font(.title3)
+            Text("Category: \(act.category.rawValue.capitalized)").foregroundColor(.secondary)
             Text("Options").font(.title).padding(.top)
             // Option selection
             ScrollView {
                 VStack(alignment: .leading) {
-                    // TODO: Correctly update fields from Model files when user inputs data
+                    // TODO: Loop through the activity options array...
```
Comment (ApolloZhu) [inline Shared/View/VoteView.swift:54]: As you might have noticed, that's not an array, but rather a dictionary (or Map in Java). My recommendation is to make the keys of the dictionary an array to ForEach on, so something like

```swift
// somewhere else in the struct where you define the other variables:
var optionIDs: [Option.ID] {
   act.options.keys.sorted() // can sort in some other ways
}

// inside the body:
ForEach(optionIDs, id: \.self) { optionID in
   ...
}
```

File: Shared/View/VoteView.swift [hunk 4]
```diff
@@ -60,6 +82,7 @@ struct VoteView: View {
 
 struct VoteView_Previews: PreviewProvider {
     static var previews: some View {
-        VoteView()
+        // TODO: Need to create ActicityID and OptionID, store in variable, and pass it in here
+        VoteView().environmentObject(UserViewModel()).environmentObject(ActivityViewModel(activityID: "AhdjSLOlTJqb68aXBAWn")).environmentObject(OptionViewModel(optionID: "B5uYuFf31aEqH5sGHFA4"))
```
Comment (ApolloZhu) [inline Shared/View/VoteView.swift:86]: Yeah, most likely you'd want VoteView to take the activity ID as a parameter and construct ActivityViewModel instance based on that. You may need to write a custom initializer (i.e. constructor)

File: Shared/View/VoteView.swift [hunk 5]
```diff
@@ -8,28 +8,50 @@
...
-            Text("Activity").font(.title).padding(.bottom)
-            Text(tempActivty.title).font(.title3)
-            Text("Category: \(tempActivty.category.rawValue.capitalized)").foregroundColor(.secondary)
+            // Get an activity to retrieve data (title, category, options)
+            // Not sure how to resolve Static method 'buildBlock' requires
+            // that 'Task<(), Never>' conform to 'View' without using Button...
+            Button {
+                Task {
+                    do {
+                        act = try await ActivityViewModel.activity(id: "AhdjSLOlTJqb68aXBAWn")
```
Comment (ApolloZhu) [inline Shared/View/VoteView.swift:30]: If you really want something to happen automatically when the view appears, use `onAppear` modifier

File: Shared/View/CreateActivityView.swift [hunk 6]
```diff
@@ -54,16 +66,33 @@ struct CreateActivityView: View {
             Spacer()
             HStack {
                 Spacer()
-                NavigationLink(destination: VoteView()) {
-                    Text("Create")
+                // Clicking on button will send info to database
+                Button {
+                    Task {
+                        do {
+                            _ = try await ActivityViewModel.createActivity(
```
Comment (ApolloZhu) [inline Shared/View/CreateActivityView.swift:73]: You probably don't want to discard the ActivityID, since you'll need it for the VoteView

**Review by ApolloZhu — CHANGES_REQUESTED** [review]
Let me know if my comments help in addressing your challenges

**Comment by anrattanasouk** [comment]
Close interaction

**Pull request closed without merging**